eql/etc/test_folding.toml

```diff
@@ -0,0 +1,452 @@
+
+[string_case_sensitive_match_comparison]
+description = "Fold string comparisons"
+case_sensitive = true
+
+   [[string_case_sensitive_match_comparison.fold.tests]]
+   expression = '"Foo" == "foo"'
+   expected = false
+
+   [[string_case_sensitive_match_comparison.fold.tests]]
+   expression = '"Foo" != "foo"'
+   expected = true
+
+
+[string_case_insensitive_match_comparison]
+description = "Fold string comparisons"
+case_sensitive = false
+
+    [[string_case_insensitive_match_comparison.fold.tests]]
+    expression = '"Foo" == "Bar"'
+    expected = false
+
+    [[string_case_insensitive_match_comparison.fold.tests]]
+    expression = '"Foo" != "Bar"'
+    expected = true
+
+    [[string_case_insensitive_match_comparison.fold.tests]]
+    expression = '"Foo" == "foo"'
+    expected = true
+
+    [[string_case_insensitive_match_comparison.fold.tests]]
+    expression = '"Foo" != "foo"'
+    expected = false
+
+
+[arithmetic_comparison]
+description = "Fold arithmetic expressions cast as booleans"
+
+    [[arithmetic_comparison.fold.tests]]
+    expression = '(1 * 2 + 3 * 4 + 10 / 2) == 19'
+    expected = true
+
+    [[arithmetic_comparison.fold.tests]]
+    expression = '(1 * 2 + 3 * 4 + 10 / 2) == (2 + 12 + 5)'
+    expected = true
+
+    [[arithmetic_comparison.fold.tests]]
+    expression = '1 * 2 + 3 * 4 + 10 / 2 == 2 + 12 + 5'
+    expected = true
+
+
+[in_set]
+description = "Fold constant check in set"
+
+    [[in_set.fold.tests]]
+    expression = "345 in (123, 345)"
+    expected = true
+
+    [[in_set.fold.tests]]
+    expression = "345 not in (123, 456)"
+    expected = true
+
+    [[in_set.fold.tests]]
+    expression = "345 in (123, 456)"
+    expected = false
+
+    [[in_set.fold.tests]]
+    expression = "'foo' not in ('foo', 'bar', 'baz')"
+    expected = false
+
+    [[in_set.fold.tests]]
+    expression = "'foo' in ('foo', 'bar', 'baz')"
+    expected = true
+
+    [[in_set.fold.tests]]
+    expression = "'foo' in ('bar', 'baz')"
+    expected = false
+
+[or]
+description = "Check that three-value boolean logic works for `or`."
+
+    [[or.fold.tests]]
+    expression = "true or false"
+    expected = true
+    
+    [[or.fold.tests]]
+    expression = "true or null"
+    expected = true
+    
+    [[or.fold.tests]]
+    expression = "true or true"
+    expected = true
+    
+    [[or.fold.tests]]
+    expression = "false or null"
+    
+    [[or.fold.tests]]
+    expression = "false or false"
+    expected = false
+    
+    [[or.fold.tests]]
+    expression = "false or true"
+    expected = true
+    
+    [[or.fold.tests]]
+    expression = "null or false"
+    
+    [[or.fold.tests]]
+    expression = "null or null"
+    
+    [[or.fold.tests]]
+    expression = "null or true"
+    expected = true
+
+[and]
+description = "Check that three-value boolean logic works for `and`."
+
+    [[and.fold.tests]]
+    expression = "true and false"
+    expected = false
+
+    [[and.fold.tests]]
+    expression = "true and null "
```

Comment:
add `# expected = null` to these to be more explicit as well as consistent with other test toml files

Reply:
```suggestion
    # expected = null
```